frontend/cypress/e2e/revision.spec.ts

```diff
@@ -0,0 +1,113 @@
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { testNoteId } from '../support/visit-test-editor'
+
+describe('Revision modal', () => {
+  const testTitle = 'testContent'
+  const testContent = `---\ntitle: ${testTitle}\n---\nThis is some test content`
+
+  beforeEach(() => {
+    cy.visitTestNote()
+    cy.intercept('GET', `api/private/notes/${testNoteId}/revisions`, [
+      {
+        id: 1,
+        createdAt: '2021-12-29T17:54:11.000Z',
+        length: 2788,
+        authorUsernames: [],
+        anonymousAuthorCount: 4,
+        title: 'Features',
+        description: 'Many features, such wow!',
+        tags: ['hedgedoc', 'demo', 'react']
+      },
+      {
+        id: 0,
+        createdAt: '2021-12-21T16:59:42.000Z',
+        length: 2782,
+        authorUsernames: [],
+        anonymousAuthorCount: 2,
+        title: 'Features',
+        description: 'Many more features, such wow!',
+        tags: ['hedgedoc', 'demo', 'react']
+      }
+    ])
+    cy.getByCypressId('sidebar.revision.button').click()
+  })
+
+  it('working', () => {
+    const cardsContents = ['Wednesday, 29 December 2021 23:24', 'Length: 2788', 'Anonymous authors or guests: 4']
+
+    cardsContents.forEach((content) => cy.get('.list-group').contains(content))
+    cy.getByCypressId('revision.modal.delete.button').should('be.visible')
+    cy.getByCypressId('revision.modal.revert.button').should('be.disabled')
+    cy.getByCypressId('revision.modal.download.button').should('be.disabled')
+  })
+
+  it('working delete revision', () => {
+    cy.getByCypressId('revision.modal.delete.button').click()
+    cy.getByCypressId('revision.delete.modal').should('be.visible')
+    cy.intercept('DELETE', `api/private/notes/${testNoteId}/revisions`, {
+      statusCode: 204
+    })
+
+    cy.getByCypressId('revision.delete.button').click()
+    cy.getByCypressId('revision.delete.modal').should('not.exist')
+  })
+  it('failed delete revision', () => {
+    cy.getByCypressId('revision.modal.delete.button').click()
+    cy.getByCypressId('revision.delete.modal').should('be.visible')
+    cy.intercept('DELETE', `api/private/notes/${testNoteId}/revisions`, {
+      statusCode: 400
+    })
+
+    cy.getByCypressId('revision.delete.button').click()
+    cy.getByCypressId('revision.delete.modal').should('not.exist')
+    cy.getByCypressId('notification-toast').should('be.visible')
+  })
+  it('working download', () => {
+    cy.intercept('GET', '/api/private/notes/mock-note/revisions/1', {
+      id: 1,
+      createdAt: '2021-12-29T17:54:11.000Z',
+      title: 'Features',
+      description: 'Many more features, such wow!',
+      tags: ['hedgedoc', 'demo', 'react'],
+      edits: [],
+      length: 2788,
+      authorUsernames: [],
+      anonymousAuthorCount: 4,
+      patch: testContent,
+      content: testContent
+    })
+
+    cy.get('.list-group').contains('Wednesday, 29 December 2021 23:24').click()
```

Comment:
The CI isn't running with the same date settings as your local computer. That's probably why it's mismatching here. Try this instead https://stackoverflow.com/a/65686935